src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -56,14 +64,23 @@ class CassandraJournal(cfg: Config) extends AsyncWriteJournal with CassandraReco
   def asyncWriteMessages(messages: Seq[AtomicWrite]): Future[Seq[Try[Unit]]] = {
     // we need to preserve the order / size of this sequence even though we don't map
     // AtomicWrites 1:1 with a C* insert
-    val serialized = messages.map(aw => Try {
+    // We must NOT catch serialization exceptions here because rejections will cause
+    // holes in the sequence number series and we use the sequence numbers to detect
+    // missing (delayed) events in the eventByTag query
```

Comment:
When exception is thrown from `asyncWriteMessages` or if the `Future` is completed with failure the `AsyncWriteJournal` will translate that to a `WriteMessageFailed` message that will cause the `PersistentActor` to stop. When it is started again it will continue from highest sequence number.

If the `Try` inside the `Seq` is a `Failure` the `AsyncWriteJournal` will translate that to a `WriteMessageRejected` and the `PersistentActor` will happily continue with next sequence number, i.e. there will be a hole.

Rejection is a new optional feature in Akka 2.4, which we don't want to support in the Cassandra plugin. http://doc.akka.io/docs/akka/2.4.1/scala/persistence.html#Failures